automerge-protocol/src/serde_impls/diff.rs

```diff
@@ -69,8 +70,7 @@ impl<'de> Deserialize<'de> for RawDiffType {
             "text" => RawDiffType::Text,
             "list" => RawDiffType::List,
             "table" => RawDiffType::Table,
-            // TODO: Not sure how to idiomatically return an error
-            _ => panic!("unexpected variant"),
+            other => return Err(Error::unknown_variant(other, VARIANTS)),
```

Comment:
Rather than using return it's probably neater to just return the result of the `match`, like

```rust
match raw_type.as_str() {
    "text" => Ok(RawDiffType::Text),
    ...
    other => Err(Error::unknown_variant(other, VARIANTS))
}
```

Reply:
Can change -- I used the return b/c I didn't want to deal with Adding `Ok` to each arm